recipes/botan/all/conanfile.py

```diff
@@ -117,84 +125,148 @@ def config_options(self):
 
         # --single-amalgamation option is no longer available
         # See also https://github.com/randombit/botan/pull/2246
-        if tools.Version(self.version) >= '2.14.0':
+        if Version(self.version) >= '2.14.0':
             del self.options.single_amalgamation
 
+        # Support for the OpenSSL provider was removed in 2.19.2
+        if Version(self.version) >= '2.19.2':
+            del self.options.with_openssl
+
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def requirements(self):
         if self.options.with_bzip2:
             self.requires("bzip2/1.0.8")
-        if self.options.with_openssl:
-            self.requires("openssl/1.1.1o")
+        if self.options.get_safe('with_openssl', False):
+            self.requires("openssl/[>=1.1 <3]")
         if self.options.with_zlib:
-            self.requires("zlib/1.2.12")
+            self.requires("zlib/[>=1.2.11 <2]")
         if self.options.with_sqlite3:
             self.requires("sqlite3/3.38.5")
         if self.options.with_boost:
-            self.requires("boost/1.79.0")
+            self.requires("boost/1.83.0")
 
     @property
     def _required_boost_components(self):
         return ['coroutine', 'system']
 
+    @property
+    def _min_cppstd(self):
+        # From the same links as below
+        return 11 if Version(self.version) < "3.0.0" else 20
+
+    @property
+    def _compilers_minimum_version(self):
+        if Version(self.version).major < 3:
+            # From https://github.com/randombit/botan/blob/2.19.3/doc/support.rst
+            return {
+                "gcc": "4.8",
+                "clang": "3.5",
+                "Visual Studio": "14",
+                "msvc": "190",
+            }
+        else:
+            # From https://github.com/randombit/botan/blob/master/doc/support.rst
+            return {
+                "gcc":  "11.2",
+                "clang": "14",
+                "apple-clang": "14",
+                "Visual Studio": "17",
+                "msvc": "193",
+            }
+
     def validate(self):
         if self.options.with_boost:
             miss_boost_required_comp = any(getattr(self.options['boost'], 'without_{}'.format(boost_comp), True) for boost_comp in self._required_boost_components)
             if self.options['boost'].header_only or self.options['boost'].shared or self.options['boost'].magic_autolink or miss_boost_required_comp:
```

Comment:
I tried to build locally and it does not work when boost is required:

```
$ conan create all --version=3.2.0 -s compiler.cppstd=20 --build=missing -pr gcc12.profile 

======== Exporting recipe to the cache ========
botan/3.2.0: Exporting package recipe: /home/conan/project/all/conanfile.py
botan/3.2.0: exports: File 'conandata.yml' found. Exporting it...
botan/3.2.0: Calling export_sources()
botan/3.2.0: Copied 1 '.yml' file: conandata.yml
botan/3.2.0: Copied 1 '.py' file: conanfile.py
botan/3.2.0: Exported to cache folder: /home/conan/.conan2/p/botan759963310d11c/e
botan/3.2.0: Exported: botan/3.2.0#9b62638872488502e0160f97996e959f (2024-01-15 15:43:57 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=12
os=Linux
[options]
botan/3.2.0:with_boost=True
botan/3.2.0:with_bzip2=True
botan/3.2.0:with_sqlite3=True
botan/3.2.0:with_zlib=True

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=12
os=Linux


======== Computing dependency graph ========
bzip2/1.0.8: Not found in local cache, looking in remotes...
bzip2/1.0.8: Checking remote: conancenter
bzip2/1.0.8: Downloaded recipe revision 457c272f7da34cb9c67456dd217d36c4
zlib/1.3: Not found in local cache, looking in remotes...
zlib/1.3: Checking remote: conancenter
zlib/1.3: Downloaded recipe revision 06023034579559bb64357db3a53f88a4
sqlite3/3.38.5: Not found in local cache, looking in remotes...
sqlite3/3.38.5: Checking remote: conancenter
sqlite3/3.38.5: Downloaded recipe revision 4b875d4249cdfb4c1235e6b3ea6c18e7
boost/1.83.0: Not found in local cache, looking in remotes...
boost/1.83.0: Checking remote: conancenter
boost/1.83.0: Downloaded recipe revision 658a6b1d7aa62a019cbba3f4ae363cb7
libbacktrace/cci.20210118: Not found in local cache, looking in remotes...
libbacktrace/cci.20210118: Checking remote: conancenter
libbacktrace/cci.20210118: Downloaded recipe revision ec1aa63bbc10145c6a299e68e711670c
b2/4.10.1: Not found in local cache, looking in remotes...
b2/4.10.1: Checking remote: conancenter
b2/4.10.1: Downloaded recipe revision 1b290403d8648c79f468f5a6496f829a
Graph root
    cli
Requirements
    boost/1.83.0#658a6b1d7aa62a019cbba3f4ae363cb7 - Downloaded (conancenter)
    botan/3.2.0#9b62638872488502e0160f97996e959f - Cache
    bzip2/1.0.8#457c272f7da34cb9c67456dd217d36c4 - Downloaded (conancenter)
    libbacktrace/cci.20210118#ec1aa63bbc10145c6a299e68e711670c - Downloaded (conancenter)
    sqlite3/3.38.5#4b875d4249cdfb4c1235e6b3ea6c18e7 - Downloaded (conancenter)
    zlib/1.3#06023034579559bb64357db3a53f88a4 - Downloaded (conancenter)
Build requirements
    b2/4.10.1#1b290403d8648c79f468f5a6496f829a - Downloaded (conancenter)
Resolved version ranges
    zlib/[>=1.2.11 <2]: zlib/1.3

======== Computing necessary packages ========
boost/1.83.0: Checking 9 compatible configurations
boost/1.83.0: Compatible configurations not found in cache, checking servers
boost/1.83.0: '1f4d8d9bfb906fd36e9471d4ce2424cf0278c2ce': compiler.cppstd=11
boost/1.83.0: 'f331bab124723d370f6d6c8b733ba3ff9df40985': compiler.cppstd=gnu11
boost/1.83.0: '0444dbb02c7c03e0924add655328614c4069992c': compiler.cppstd=14
boost/1.83.0: '53fadd80e75df07d899e7a29532a2fc59cd7ab23': compiler.cppstd=gnu14
boost/1.83.0: '21c1ffdf955757b3eaf1428692e72e91c15d8446': compiler.cppstd=17
boost/1.83.0: 'ab3b66987b0ff007c45a5b90708ec0b68208653b': compiler.cppstd=gnu17
boost/1.83.0: 'b7691adcb83380a1f2a05c4982136fd2dcb71ab7': compiler.cppstd=gnu20
boost/1.83.0: 'b4575a8b8adc07f0f83596aa71fc38363b522735': compiler.cppstd=23
boost/1.83.0: '1d726159589d984732e608f539092dd02c6b8583': compiler.cppstd=gnu23
ERROR: botan/3.2.0: Error in validate() method, line 181
	miss_boost_required_comp = any(getattr(self.options['boost'], 'without_{}'.format(boost_comp), True) for boost_comp in self._required_boost_components)
while calling '<genexpr>', line 181
	miss_boost_required_comp = any(getattr(self.options['boost'], 'without_{}'.format(boost_comp), True) for boost_comp in self._required_boost_components)
	ConanException: option 'without_coroutine' doesn't exist
Possible options are []

```

So, doing a quick check, in Conan 2.x you should use `self.dependencies["boost"].options` instead. Could you please update and check it?